api/domain_handlers.go

```diff
@@ -387,8 +387,12 @@ func (api API) Remove(w http.ResponseWriter, r *http.Request) {
 // database and change the status to PendingAutomatedRemoval if the domain
 // does not follow the requirements for more than 2 crawls
 
-// Example: GET /removeineligibledomains?
+// Example: GET /remove-ineligible-domains
 func (api API) RemoveIneligibleDomains(w http.ResponseWriter, r *http.Request) {
+	// ensures endpoint requests can only come from App Engine
+	if r.Header.Get("X-Appengine-Cron") != "true" || r.RemoteAddr != "0.1.0.2" {
+		return
```

Comment:
Before returning, at least set a status code on the response indicating that the request won't be processed (e.g. a 403).

Also, add a test case that if the header or ip address is missing, the status code gets returned.

Reply:
Done

Reply:
nit: use a constant (`http.StatusForbidden`) instead of a hardcoded value

Reply:
Done